### PR TITLE
Add Enterprise SMS Report

### PR DESCRIPTION
**Pull request by mjriley**
## Product Description
Adds a new Enterprise SMS Report, including associated tile and API.

## Technical Summary
Associated ticket: https://dimagi.atlassian.net/browse/SAAS-16131

At the moment, the new API is implemented using the old population logic, as the new iterator logic was only written for Elasticsearch queries and didn't take into account how to generate summary statistics. The intent is to eventually use the new iterator logic, but more work will need to be done to determine how best to support the SMS queries with that framework

## Feature Flag
We mentioned we would prefer to do an iterative approach to these new tiles, so this is ready for release without a feature flag

### Safety story
Tested locally -- generated the report, accessed the API through the browser, and accessed it through PowerBI. Verified that UI worked as expected.

### Automated test coverage

None at the moment. As it stands, the code to be tested is the `EnterpriseSMSReport` code, but I'd like to eventually move the logic out of there into a query class and test against that.

### QA Plan
TBD

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/enterprise/enterprise.py
Comment (jingcheng16) [inline corehq/apps/enterprise/enterprise.py:394]: Change the title to align with the design doc

File: corehq/apps/enterprise/enterprise.py
Comment (jingcheng16) [inline corehq/apps/enterprise/enterprise.py:400]: Can you make it as a util function so other tiles can reuse it?

File: corehq/apps/enterprise/enterprise.py
Comment (jingcheng16) [inline corehq/apps/enterprise/enterprise.py:434]: Simplify the header to be `Project Space`

File: corehq/apps/enterprise/enterprise.py
Comment (jingcheng16) [inline corehq/apps/enterprise/enterprise.py:442]: Should we only include sms that is succesfully sent and received? If we don't charge them, then we should not include errored sms in `num_sent` and `num_received`

Comment (jingcheng16) [reply corehq/apps/enterprise/enterprise.py:442]: `total_for_domain` should also align with the number in report, when people sum up num_sent for all domains

File: corehq/apps/enterprise/api/resources.py
Comment (jingcheng16) [inline corehq/apps/enterprise/api/resources.py:327]: util function for it?

**Review by jingcheng16 — CHANGES_REQUESTED** [review]
Reviewed partial of the PR with @biyeun

File: corehq/apps/enterprise/enterprise.py
Comment (mjriley) [reply corehq/apps/enterprise/enterprise.py]: Addressed in fc4fccdf5a1f35f682deee56395310c127614c83

File: corehq/apps/enterprise/enterprise.py
Comment (mjriley) [reply corehq/apps/enterprise/enterprise.py:400]: I believe I did this in the iterators PR, which is still in review. Re-adding it here would cause conflicts, so I'm going to wait until that PR is merged for this functionality

File: corehq/apps/enterprise/enterprise.py
Comment (mjriley) [reply corehq/apps/enterprise/enterprise.py]: 8310efd046ed9a70859397466b507bf5e1b82a21

File: corehq/apps/enterprise/enterprise.py
Comment (mjriley) [reply corehq/apps/enterprise/enterprise.py]: Addressed in 1adcaf384f41033b97d3377f9185ae91c9a4562a

File: corehq/apps/enterprise/api/resources.py
Comment (mjriley) [reply corehq/apps/enterprise/api/resources.py]: c1816549f05dfaac9f8902e106b6a2d3896330e9

**Comment by jingcheng16** [comment]
@mjriley Can review again once the conflicts with master are resolved. Btw, do you plan to put it through QA?

File: corehq/apps/enterprise/enterprise.py
Comment (jingcheng16) [inline corehq/apps/enterprise/enterprise.py:425]: What does the `processed` attribute mean in SMS class?

Comment (mjriley) [reply corehq/apps/enterprise/enterprise.py:425]: I believe it gets set when a message completes processing -- you can see that [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/sms/tasks.py#L111-L119). One example of why we need this would be code like [this](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/sms/tasks.py#L281-L289), where the message doesn't have an `error` yet, but we are in the process of retrying it. That message shouldn't be reported yet, so we filter it out by requiring `processed` to be `True`

**Review by jingcheng16 — APPROVED** [review]

**Pull request merged**
